Request/ParamConverter/DoctrineParamConverter.php

```diff
@@ -50,8 +58,16 @@ public function apply(Request $request, ParamConverter $configuration)
             $configuration->setIsOptional(true);
         }
 
-        // find by identifier?
-        if (false === $object = $this->find($class, $request, $options, $name)) {
+        $errorMessage = null;
+        if ($expr = $options['expr']) {
+            $object = $this->findViaExpression($class, $request, $expr, $options);
+
+            if (null === $object) {
+                $errorMessage = sprintf('The expression "%s" returned null', $expr);
```

Comment:
This is used below in the `NotFoundHttpException` exception. Do we assume these messages are safe? If not, that's a bummer as it requires us to use really poor exception messages (unless we detected `debug` mode?)